velox/vector/VectorPool.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/vector/VectorPool.h"
+
+namespace facebook::velox {
+
+VectorPtr VectorPool::get(const TypePtr& type, vector_size_t size) {
+  auto kind = static_cast<int32_t>(type->kind());
+  if (kind < kNumCachedVectorTypes && size <= kMaxRecycleSize) {
+    return vectors_[kind].pop(type, size, *pool_);
+  }
+  return BaseVector::create(type, size, pool_);
+}
+
+bool VectorPool::release(VectorPtr& vector) {
+  if (!vector.unique() || vector->size() > kMaxRecycleSize) {
+    return false;
+  }
+  auto kind = static_cast<int32_t>(vector->typeKind());
```

Comment:
nit: how about add anonymous function say: int32_t typeKindToCacheIndex(TypeKind kind)?

Reply:
Good idea. Let me do that.